Sources/SPTDataLoaderServiceSessionSelector.m

```diff
@@ -96,8 +117,9 @@ - (NSURLSession *)createWaitingSession
 
 - (void)invalidateAndCancel
 {
-    [self.waitingSession invalidateAndCancel];
-    [self.nonWaitingSession invalidateAndCancel];
```

Comment:
if possible it would be better to have those changes in a different commit, it is ok if they are part of this PR

Reply:
I think this is a good addition. If the sessions haven't been created yet, there's no reason to create them just to invalidate them.